api/rosetta/data_network_integration_test.go

```diff
@@ -0,0 +1,523 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// +build integration
+
+package rosetta_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/optakt/flow-dps/api/rosetta"
+	"github.com/optakt/flow-dps/models/convert"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/configuration"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+)
+
+func TestAPI_Status(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	const (
+		lastHeight    = 425
+		oldestBlockID = "d47b1bf7f37e192cf83d2bee3f6332b0d9b15c0aa7660d1e5322ea964667b333"
+	)
+
+	lastBlock := knownHeader(425)
+
+	request := rosetta.StatusRequest{
+		NetworkID: defaultNetwork(),
+	}
+
+	rec, ctx, err := setupRecorder(statusEndpoint, request)
+	require.NoError(t, err)
+
+	err = api.Status(ctx)
+	assert.NoError(t, err)
+
+	assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
+
+	// unpack response
+	var status rosetta.StatusResponse
+	require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &status))
+
+	// verify current block
+	assert.Equal(t, status.CurrentBlockID.Index, lastBlock.Height)
+	assert.Equal(t, status.CurrentBlockID.Hash, lastBlock.ID().String())
+	assert.Equal(t, status.CurrentBlockTimestamp, convert.RosettaTime(lastBlock.Timestamp))
+
+	// verify oldest block
+	assert.Equal(t, status.OldestBlockID.Hash, oldestBlockID)
+
+	// this is actually omitted from JSON, due to height being zero, and JSON having the 'omitempty' tag
```

Comment:
That's an edge case I didn't think would actually come up; on the main network, we will probably never have a zero height, as the root block is too far in incompatible sporks. So when height is zero, we assume it's empty and use the hash as reference.

However, it's true that if we want it to properly work on test networks, we should support having zero. So I think we should change the `Index` in the `identifier.Block` to a `*int` that is `nil` when not set, and `*Index == 0` when zero.

Reply:
Will we ever return anything with the `Height`/`Index` unset? If not, we could just remove the `omitempty` part of the tag, no?

Reply:
We won't ever return anything, but we might receive. So when we receive it, there should be a way for the consumer to express "I want block at height 0" and "I don't want to specify a block height" as distinct cases.

Reply:
Aaah, right. Good point. 👀 

Should I create an issue for it?

Reply:
Yes, of course.